test/distributed/pipelining/test_microbatch.py

```diff
@@ -59,6 +60,120 @@ def test_split_and_merge(self):
         torch.testing.assert_close(merged_kwargs, kwargs)
         print("Microbatch test passed")
 
+    def test_split_block_mask(self, device):
+        B = 6
+        H = 1
+        SEQ_LEN = 512
+        DIM = 32
+        DOC_LEN = 30
+
+        def create_block_causal_mask(batch, eos_id: int):
+            mask = batch == eos_id
+            mask[:, -1] = True
+            acc_mask = torch.cumsum(torch.where(mask, 1, 0), dim=1)
+            seq_idx = torch.zeros_like(acc_mask, dtype=torch.int32)
+            seq_idx[:, 1:] = acc_mask[:, :-1]
+
+            def block_causal_mask(
+                b: torch.Tensor,
+                h: torch.Tensor,
+                q_idx: torch.Tensor,
+                kv_idx: torch.Tensor,
+            ):
+                return (seq_idx[b, q_idx] == seq_idx[b, kv_idx]) & (q_idx >= kv_idx)
+
+            return block_causal_mask
+
+        batch = list(range(DOC_LEN)) * 1024
+        batch = torch.tensor(batch[: B * SEQ_LEN], device=device).reshape(B, SEQ_LEN)
```

Comment:
nit: this is a little weird to me, I think at least do comment or  have some assert that doc_len * 1024 > b * seqlen